File: Human_written_code/Code_with_comments/Python/Q27.py
# Read input values from standard input
def main():
    # Number of vectors
    n = int(input().strip())
    
    # Initialize lists to store x, y, z coordinates
    x = []
    y = []
    z = []
    
    # Read each vector's coordinates
    for _ in range(n):
        xi, yi, zi = map(int, input().split())
        x.append(xi)
        y.append(yi)
        z.append(zi)
    
    # Calculate the sum of each coordinate axis
    sum1 = sum(x)
    sum2 = sum(y)
    sum3 = sum(z)
    
    # Check if all sums are zero
    if sum1 == 0 and sum2 == 0 and sum3 == 0:
        print("YES")
    else:
        print("NO")

if __name__ == "__main__":
    main()

File: Human_written_code/Code_with_comments/CPP/Q27.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;  // Read the number of force vectors
    int x[n], y[n], z[n];

    // Read the components of each vector
    for (int i = 0; i < n; i++) {
        cin >> x[i] >> y[i] >> z[i];
    }

    int sum1 = 0, sum2 = 0, sum3 = 0;

    // Calculate the sum of components in each direction
    for (int i = 0; i < n; i++) {
        sum1 += x[i];
        sum2 += y[i];
        sum3 += z[i];
    }

    // Check if the vector sum is zero in all directions
    if (sum1 == 0 && sum2 == 0 && sum3 == 0)
        cout << "YES" << endl;  // The forces are in equilibrium
    else
        cout << "NO" << endl;   // The forces are not in equilibrium

    return 0;
}
